compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -2112,6 +2368,118 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             );
         }
     }
+
+    fn emit_generic_mismatches(
```

Comment:
label_generic_mismatches is a better name i think

Reply:
Agreed